Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -38,23 +38,54 @@
             "name": "PostDeploymentAction",
             "displayName": "Post Deployment Action",
             "isExpanded": false,
-            "visibleRule": "WebAppKind != \"\""
+            "visibleRule": "ConnectionType = AzureRM || WebAppKind != \"\""
         },
         {
             "name": "ApplicationAndConfigurationSettings",
             "displayName": "Application and Configuration Settings",
-            "isExpanded": false
+            "isExpanded": false,
+            "visibleRule": "ConnectionType = AzureRM"
         }
     ],
     "inputs": [
+        {
+            "name": "ConnectionType",
+            "type": "pickList",
+            "label": "Connection Type",
+            "defaultValue": "AzureRM",
+            "options": {
+                "AzureRM": "Azure Resource Manager",
+                "PublishProfile": "Publish Profile"
+            },
+            "required": true
+        },
         {
             "name": "ConnectedServiceName",
             "aliases": ["azureSubscription"],
             "type": "connectedService:AzureRM",
             "label": "Azure subscription",
             "defaultValue": "",
             "required": true,
-            "helpMarkDown": "Select the Azure Resource Manager subscription for the deployment."
+            "helpMarkDown": "Select the Azure Resource Manager subscription for the deployment.",
+            "visibleRule": "ConnectionType = AzureRM"
+        },
+        {
+            "name": "PublishProfilePath",
+            "type": "filePath",
+            "label": "Publish Profile Path",
+            "defaultValue": "$(System.DefaultWorkingDirectory)/**/*.pubxml",
+            "required": true,
+            "helpMarkDown": "",
+            "visibleRule": "ConnectionType = PublishProfile"
+        },
+        {
+            "name": "PublishProfilePassword",
```

Comment:
Can we avoid using default value here? Instead provide a helptext (like other tasks) to set password in secret variable and use the variable.

Reply:
removed default value and added helptext. I will ask Atul if changes needed in helptext.